rpcs3/Emu/Cell/Modules/cellSaveData.cpp

```diff
@@ -849,59 +859,81 @@ static NEVER_INLINE error_code savedata_op(ppu_thread& ppu, u32 operation, u32 v
 
 		u32 size_kbytes = 0;
 
+		std::vector<fs::dir_entry> files_sorted;
+
 		for (auto&& entry : fs::dir(dir_path))
 		{
 			entry.name = vfs::unescape(entry.name);
 
-			// only files, system files ignored, fileNum is limited by setBuf->fileListMax
 			if (!entry.is_directory)
 			{
 				if (entry.name == "PARAM.SFO" || entry.name == "PARAM.PFD")
 				{
 					continue; // system files are not included in the file list
 				}
 
-				statGet->fileNum++;
+				files_sorted.push_back(entry);
+			}
+		}
 
-				size_kbytes += (entry.size + 1023) / 1024; // firmware rounds this value up
+		// clang-format off
+		std::sort(files_sorted.begin(), files_sorted.end(), [&](const fs::dir_entry& a, const fs::dir_entry& b) -> bool
+		{
+			const auto a_it = std::find(blist.begin(), blist.end(), a.name);
+			const auto b_it = std::find(blist.begin(), blist.end(), b.name);
 
-				if (statGet->fileListNum >= setBuf->fileListMax)
-					continue;
+			if (a_it == blist.end() && b_it == blist.end())
+			{
+				// Order alphabetically for old saves
+				return a.name.compare(b.name);
+			}
 
-				statGet->fileListNum++;
+			return a_it < b_it;
+		});
+		// clang-format on
 
-				auto& file = *file_list++;
+		for (auto&& entry : files_sorted)
+		{
+			statGet->fileNum++;
 
-				file.size = entry.size;
-				file.atime = entry.atime;
-				file.mtime = entry.mtime;
-				file.ctime = entry.ctime;
-				strcpy_trunc(file.fileName, entry.name);
+			size_kbytes += (entry.size + 1023) / 1024; // firmware rounds this value up
```

Comment:
Try to add another {} level so this huge block doesn't appear as deleted and rewritten.

Reply:
Isn't that kinda unnecessary with so few lines

Reply:
done